src/vs/workbench/electron-browser/actions.ts

```diff
@@ -918,4 +922,289 @@ export class ToggleSharedProcessAction extends Action {
+		return this.panelService.openPanel(activePanelId, true);
+	}
+
+	protected navigateToSidebar(): TPromise<any> {
+		if (!this.partService.isVisible(Parts.SIDEBAR_PART)) {
+			return TPromise.as(false);
+		}
+
+		const activeViewletId = this.viewletService.getActiveViewlet().getId();
+		return this.viewletService.openViewlet(activeViewletId, true);
+	}
+
+	protected navigateToLastActiveEditor(): TPromise<any> {
+		const model = this.groupService.getStacksModel();
+		if (model.groups.length < 1) {
+			return null;
+		}
+
+		const lastGroup = model.activeGroup;
+		const activeEditor = lastGroup.activeEditor;
+		return this.editorService.openEditor(activeEditor, null, model.positionOfGroup(lastGroup));
+	}
+
+	protected navigateToStartOfEditor(): TPromise<any> {
+		const model = this.groupService.getStacksModel();
+		if (model.groups.length < 1) {
+			return null;
+		}
+
+		const firstGroup = model.getGroup(0);
+		const firstEditor = firstGroup.getEditor(0);
+		return this.editorService.openEditor(firstEditor, null, model.positionOfGroup(firstGroup));
+	}
+
+	protected navigateToEndOfEditor(): TPromise<any> {
+		const model = this.groupService.getStacksModel();
+		if (model.groups.length < 1) {
+			return null;
+		}
+
+		const lastGroup = model.getGroup(model.groups[model.groups.length - 1].id);
+		const lastEditor = lastGroup.getEditor(lastGroup.count - 1);
+		return this.editorService.openEditor(lastEditor, null, model.positionOfGroup(lastGroup));
+	}
+}
+
+export class NavigateLeftAction extends BaseNavigationAction {
+
+	public static ID = 'workbench.action.navigateLeft';
+	public static LABEL = nls.localize('navigateLeft', "Navigate Left");
+
+	constructor(
+		id: string,
+		label: string,
+		@IWorkbenchEditorService editorService: IWorkbenchEditorService,
+		@IEditorGroupService groupService: IEditorGroupService,
+		@IPanelService panelService: IPanelService,
+		@IPartService partService: IPartService,
+		@IViewletService viewletService: IViewletService
+	) {
+		super(id, label, editorService, groupService, panelService, partService, viewletService);
+	}
+
+	protected navigateOnEditorFocus(isEditorGroupVertical, isSidebarPositionLeft): TPromise<boolean> {
+		if (isEditorGroupVertical) {
+			return this.navigateLeftFromEditor(true, isSidebarPositionLeft);
+		}
+		return this.navigateLeftFromEditor(false, isSidebarPositionLeft);
+	}
+
+	protected navigateOnPanelFocus(isEditorGroupVertical, isSidebarPositionLeft): TPromise<boolean> {
+		if (isSidebarPositionLeft) {
+			return this.navigateToSidebar();
+		}
+		return TPromise.as(false);
+	}
+
+	protected navigateOnSidebarFocus(isEditorGroupVertical, isSidebarPositionLeft): TPromise<boolean> {
+		if (isSidebarPositionLeft) {
+			return TPromise.as(false);
+		}
+		if (isEditorGroupVertical) {
+			return this.navigateToEndOfEditor();
+		}
+		return this.navigateToLastActiveEditor();
+	}
+
+	private navigateLeftFromEditor(jumpGroups: boolean, isSidebarPositionLeft: boolean): TPromise<boolean> {
```

Comment:
Why would we ever not jump groups? I am in a state where I have 2 editors open in one group and navigating left and right jumps between those tabs. My understanding is that this is entirely not about navigating between tabs?